lib/dlopen.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const binding = process.binding('dlopen');
+
+exports.dlopen = function dlopen(name) {
+  var lib = new Buffer(binding.sizeof_uv_lib_t);
```

Comment:
This needs to be reconsidered because of the upcoming V8 changes. Here you'd essentially be using a `Uint8Array()`. Problem is that with the new implementation we are going to allow V8'd GC to handle this automatically, and there's no way to use conditional logic to make the typed array external and persistent on instantiation. Best you might be able to do is pass the new instance to another function that does this for you.

But for performance an simplifying code complexity I would instead suggest returning a `Persistent<External>`.